apps/ejabberd/src/mod_admin_extra_accounts.erl

```diff
@@ -192,73 +199,66 @@ delete_old_users_vhost(Host, Days) ->
     {ok, io_lib:format("Deleted ~p users: ~p", [N, UR])}.
 
 
--spec delete_old_users(Days :: integer(), Users :: [ejabberd:simple_jid()])
-            -> {'removed', non_neg_integer(), [ejabberd:simple_jid()]}.
+-spec delete_old_users(Days :: integer(), Users :: [ejabberd:simple_bare_jid()]) ->
+    {removed, non_neg_integer(), [ejabberd:simple_bare_jid()]}.
 delete_old_users(Days, Users) ->
     %% Convert older time
     SecOlder = Days*24*60*60,
 
     %% Get current time
     {MegaSecs, Secs, _MicroSecs} = now(),
```

Comment:
I agree with @gadgetci. Let's kill with `p1_time_compat` or whatever is suitable.